### PR TITLE
feat: statistics in json format

### PR DESCRIPTION
**Pull request by lucarin91**

### CODE REVIEW

File: stats.go [hunk 1]
```diff
@@ -1,23 +1,43 @@
...
+				atomic.LoadUint64(&s.Redirects.Success),
+				atomic.LoadUint64(&s.Redirects.Failed),
+			},
+			make([]Handler, len(s.Handlers)),
+		},
+	}
+
+	for i, v := range s.Handlers {
+		newS.Handlers[i] = Handler{v.Name, atomic.LoadUint64(&v.Count)}
+	}
```
Comment (arl) [inline stats.go:23]: here you can use the `copy` builtin to copy the slice

File: route.go [hunk 2]
```diff
@@ -47,24 +48,24 @@ func (s *Server) shorten(w http.ResponseWriter, r *http.Request) {
...
-	s.updateStats(Statistics)
-
-	fmt.Fprintf(w, "URLs: %v\n", len(s.Kvs.storage))
-
-	fmt.Fprintf(w, "Redirect: %v\n", atomic.LoadUint64(&s.Stats.Redirects.Success))
-
-	fmt.Fprintf(w, "Handler:\n")
-	for _, v := range s.Stats.Handlers {
-		fmt.Fprintf(w, "  %v: %v\n", v.Name, atomic.LoadUint64(&v.Count))
+	atomic.AddUint64(&s.Stats.Handlers[Statistics].Count, 1)
```
Comment (arl) [inline route.go:57]: bonus point: can you think of a way to use an http middleware to handle in a single point the increment of the handler count metric?

You don't have to do in this PR, if you have some time between 2 exercices though you can have a look at https://about.sourcegraph.com/go/gophercon-2019-how-i-write-http-web-services-after-eight-years/ that explains clearly some interesting and idiomatic ways to create http servers in go, and middlewares.

**Review by arl — APPROVED** [review]
Good job. To answer to your question on slack, I don't think there's a better way to do it (speaking about the use of `atomic`).

When there is metrics gathering in production code, surely these metrics are published (via an endpoint or whatever) and generally these 2 actions take place in 2 different threads/goroutines. So there's shared state to handle. Sometimes you have to use a mutex, sometimes atomic are enough.

Another thing I noted is that we should handle the error reported by http.ListenAndServe in main. On my computer I had another server running  at `:8080` and for some minutes I didn't understand why url-shortener was writing `Start` and exiting immediately :-)

File: stats.go (hunk 1 above)
Comment (lucarin91) [reply stats.go:23]: If I copy the slice with the `copy` primitive the counter will be copied atomically? If not it maybe lead to data-race. However, I think that in the x86 platform a words will be always read atomically.

Comment (arl) [reply stats.go:23]: So all in all, the use of atomic is required here for correctness.
Here a link that might be of interest about that:
https://dave.cheney.net/2018/01/06/if-aligned-memory-writes-are-atomic-why-do-we-need-the-sync-atomic-package

Also go has a race detector, which comes very handy. `go run -race` or `go build -race`

**Pull request merged**